src/app/api/submissions/latest/route.ts

```diff
@@ -0,0 +1,90 @@
+import { NextRequest, NextResponse } from 'next/server';
+
+// Helper to check if the origin is allowed
+function isAllowedOrigin(origin: string | null) {
+  if (!origin) return false;
+  try {
+    const { hostname } = new URL(origin);
+    return hostname.endsWith('.coderbase.dev') || hostname === 'coderbase.dev';
+  } catch {
+    return false;
+  }
+}
+
+// Placeholder for your actual data fetching logic
+async function getLatestSubmissionByEmail(email: string) {
+  // TODO: Replace with real data fetching (e.g., from Supabase)
+  return { id: 'example-id', email, content: 'Latest submission', createdAt: new Date().toISOString() };
+}
+
+function corsHeaders(origin: string | null): Record<string, string> {
+  if (isAllowedOrigin(origin)) {
+    return {
+      'Access-Control-Allow-Origin': origin!,
+      'Access-Control-Allow-Methods': 'GET, OPTIONS',
+      'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+    };
+  }
+  // Return an empty object, not undefined values
+  return {};
+}
+
+export async function GET(req: NextRequest) {
+  const email = req.nextUrl.searchParams.get('email');
+  const origin = req.headers.get('origin');
+
+  if (!isAllowedOrigin(origin)) {
```

Comment:
- Using per-route logic for CORS is sufficient for this case, since it allows for flexible, endpoint-specific control.
- However, if you ever want to block all requests from disallowed origins before they reach any app logic (for example, for security or performance reasons), you could also implement CORS handling in Next.js **middleware.** This isn’t required for this PR, but it’s a option to be aware of Middleware. Consider adding a note about Middleware in Next.js to your technical documentation in Google Docs nhé @Shepe1304